File: typescript-client-sdk/src/sdk/models/shared/datefilter.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



export class DateFilter extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=equals" })
  equals?: string;

  @SpeakeasyMetadata({ data: "json, name=gt" })
  gt?: string;

  @SpeakeasyMetadata({ data: "json, name=gte" })
  gte?: string;

  @SpeakeasyMetadata({ data: "json, name=in" })
  in?: string[];

  @SpeakeasyMetadata({ data: "json, name=lt" })
  lt?: string;

  @SpeakeasyMetadata({ data: "json, name=lte" })
  lte?: string;

  @SpeakeasyMetadata({ data: "json, name=not" })
  not?: string;

  @SpeakeasyMetadata({ data: "json, name=notIn" })
  notIn?: string[];
}

File: go-client-sdk/pkg/models/shared/datefilter.go
package shared

type DateFilter struct {
	Equals *string  `json:"equals,omitempty"`
	Gt     *string  `json:"gt,omitempty"`
	Gte    *string  `json:"gte,omitempty"`
	In     []string `json:"in,omitempty"`
	Lt     *string  `json:"lt,omitempty"`
	Lte    *string  `json:"lte,omitempty"`
	Not    *string  `json:"not,omitempty"`
	NotIn  []string `json:"notIn,omitempty"`
}
